rust/assistant/src/lib.rs

```diff
@@ -792,20 +793,32 @@ impl GenerateOutput {
 
     /// Create a `SuggestionInlineType` from the output that can be used for the `suggestion`
     /// property of the instruction
-    pub fn to_suggestion_inline(self) -> SuggestionInlineType {
-        SuggestionInlineType::InsertInline(InsertInline {
-            content: self.nodes.into_inlines(),
+    pub fn to_suggestion_inline(self, insert: bool) -> SuggestionInlineType {
+        if insert {
+            return SuggestionInlineType::InsertInline(InsertInline {
+                content: self.nodes.into_inlines(),
+                ..Default::default()
+            });
+        }
+        SuggestionInlineType::ReplaceInline(ReplaceInline {
+            replacement: self.nodes.into_inlines(),
             ..Default::default()
         })
```

Comment:
Avoid early returns and instead just "return" the if expression:

```suggestion
        if insert {
            SuggestionInlineType::InsertInline(InsertInline {
                content: self.nodes.into_inlines(),
                ..Default::default()
            })
        } else {
            SuggestionInlineType::ReplaceInline(ReplaceInline {
                replacement: self.nodes.into_inlines(),
                ..Default::default()
            })
       }
```